### PR TITLE
feat: implement gaussian naive bayes

### PR DESCRIPTION
**Pull request by rouille**
### Purpose
Implement Gaussian naive Bayes algorithm.

### What the code is doing
* The `NaiveBayes` base class stores the training examples, groups training examples by class and calculate the prior, $P(y)$, of each class as the relative frequency of class $y$ in training set.
* The `GaussianNaiveBayes` child class classifies test examples using the following Maximum A Posteriori (MAP) estimation: $\hat{y} = argmax_y P(y) \prod_i P(x_i|y)$ or $\log P(y) + \sum_i log P(x_i|y)$, where $P(x_i|y)$ is the likelihood of feature $x_i$ that is assumed to be Gaussian. This classification rule is derived from Bayes theorem with the naive assumption of conditional independence between every pair of features given the value of the class variable.

### Testing
Compare implemented algorithm to the one from `scikit-learn` using the **iris** dataset.

### Where to look
The `interview.algorithm.naive_bayes` module encloses the base and child classes

### Usage Example/Visuals
N/A

### Time estimate
10min

### CODE REVIEW

**Review by kasparm — APPROVED** [review]
This works for me.
If the goal is to make this class more use friendly you would implement error handling :)

File: interview/algorithm/naive_bayes.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        self.x_train = x_train
+        self.y_train = y_train
+        self.classes = list(set(y_train))
+
+    def arrange(self):
+        """Group training examples by class
+
+        :return: (*dict*) -- dictionary giving training examples for each class.
+        """
+        return {c: self.x_train[np.where(self.y_train == c)] for c in self.classes}
```
Comment (BainanXia) [inline interview/algorithm/naive_bayes.py:33]: Another way to do this is using `itertools.groupby`

File: interview/algorithm/naive_bayes.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        :param list x_test: test dataset. Each test example has the same number of
+            features than a training example.
+        :raises NotImplementedError: method is implemented in child classes.
+        """
+        raise NotImplementedError("Implemented in child classes")
+
+
+class GaussianNaiveBayes(NaiveBayes):
+    """mplements the Gaussian Naive Bayes algorithm for classification. The
```
Comment (BainanXia) [inline interview/algorithm/naive_bayes.py]: mplements -> implements

File: interview/algorithm/naive_bayes.py (hunk 1 above)
Comment (rouille) [reply interview/algorithm/naive_bayes.py:33]: Yes but I am not sure it would be shorter.

File: interview/algorithm/naive_bayes.py (hunk 2 above)
Comment (rouille) [reply interview/algorithm/naive_bayes.py]: Done

**Review by BainanXia — APPROVED** [review]
LGTM

**Pull request merged**